### PR TITLE
Twitter Communication methods

### PR DESCRIPTION
**Pull request by DanielEizenberger**

### CODE REVIEW

**Review by Germiee — APPROVED** [review]
Fetch tweet could possibly be randomized so that the tweets don't get tweeted in a specific order, but for now this is fine. 💯 🔥 👍 🥇

File: src/twitter_communications.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+def get_new_messages(client) :
+    logger.info("Trying to get direct messages")
+    try :
+        all_messages = client.get_direct_messages(count = 200)
+
+        logger.info("Number of messages (Including the bots messages): {}", len(all_messages))
+
+        new_messages = []
+        for mes in all_messages:
+            if mes.message_create["sender_id"] != "1332712271987023874" :
```
Comment (DanielEizenberger) [inline src/twitter_communications.py]: Please replace the string ("1332712271987023874") with a global variable

File: src/twitter_communications.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+def send_message(client, id, content):
+    logger.info("Sending direct message to the user with the ID: {}", id)
+    try :
+        client.send_direct_message(id,content)
+    except Exception as e:
+        logger.error("Could not send message")
+        if hasattr(e, 'message'):
+            logger.error("Error message: {}", e.message)
+        else:
+            logger.error("Error message: {}", e)
```
Comment (DanielEizenberger) [inline src/twitter_communications.py]: New line is missing

File: src/twitter_communications.py [hunk 3]
```diff
@@ -85,4 +85,17 @@ def send_message(client, id, content):
...
+    try : 
+        user = client.get_user(user_id)
+        logger.info(user.name)
+        return user.name # should return the users name, depends on twitters user object which is weird
+    except Exception as e:
+        logger.error("Could not get username")
+        if hasattr(e, 'message'):
+            logger.error("Error message: {}", e.message)
+        else:
+            logger.error("Error message: {}", e)
```
Comment (DanielEizenberger) [inline src/twitter_communications.py]: New line is missing

**Comment by Germiee** [comment]
We should squash this, 45 commits with often many small testing based changes are not good to have on our main branch and only cause utter confusion

Main reason for this was the API keys etc so this shouldn't be an issue anymore, atleast not to this extent

**Pull request merged**